### PR TITLE
Update GPU-Targets.rst

### PR DESCRIPTION
**Pull request by huanzhang12**
I found that there are some misleading information in GPU-Targets.rst. I removed some examples provided by @Laurae2 because the provided parameters only work for a specific machine and a specific OpenCL driver version, and can mislead users like #1608

However I am not able to do any tests on Windows platform at this time, especially that AMD APP SDK is now deprecated, and I am not exactly sure about the current OpenCL status with AMD on Windows. I think the OpenCL runtime are now provided by AMD drivers rather than AMD APP SDK. I will really appreciate it if someone can test LightGBM on Windows with AMD GPUs without AMD APP SDK; is it enough to just install drivers from AMD?

### CODE REVIEW

File: docs/GPU-Targets.rst [hunk 1]
```diff
@@ -4,39 +4,42 @@ GPU SDK Correspondence and Device Targeting Table
...
 
 --------------
 
-Targeting Table
-===============
+Query OpenCL Devices in Your System
+===================================
+
+Your system might have multiple GPUs from different vendors ("platforms") installed. Setting up LightGBM GPU device requries two parameters: OpenCL Platform ID (``gpu_platform_id``) and OpenCL Device ID (``gpu_device_id``). Generally speaking, each vendor provides a OpenCL Platform, and devices from the same vendor have different device IDs under that platform. For example, if your system has a Intel Integrated GPU and two discrete GPUs from AMD, you will have two OpenCL platforms (with ``gpu_platform_id=0`` and ``gpu_platform_id=1``). If the platform 0 is Intel, it has one device (``gpu_device_id=0``) representing the Intel GPU; If the platform 1 is AMD, it has two devices (``gpu_device_id=0``, ``gpu_device_id=1``) representing the two AMD GPUs. If you have a discrete GPU by AMD/NVIDIA and an integrated GPU by Intel, make sure to select the correct ``gpu_platform_id`` to use the discrete GPU as it usually provides better performance.
+
```
Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: "has ``a`` Intel Integrated GPU" -> "has Intel ``i``ntegrated GPU""

Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: "Intel GPU; If the platform 1" -> "Intel GPU; `i`f the platform 1"

Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: I think it's good idea to make some cross links for parameters list.
In `GPU-Targets.rst`:
```
... requries two parameters: `OpenCL Platform ID <./Parameters.rst#gpu_platform_id>`__ (``gpu_platform_id``) and `OpenCL Device ID <./Parameters.rst#gpu_device_id>`__ (``gpu_device_id``).
```
and in [`config.h`](https://github.com/Microsoft/LightGBM/blob/master/include/LightGBM/config.h) (after modifying config.h please run `python ../../helper/parameter_generator.py`):
```
// desc = OpenCL platform ID. Usually each GPU vendor exposes one OpenCL platform
// desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details

...

// desc = OpenCL device ID in the specified platform. Each GPU in the selected platform has a unique device ID
// desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details
```

File: docs/GPU-Targets.rst [hunk 2]
```diff
@@ -4,39 +4,42 @@ GPU SDK Correspondence and Device Targeting Table
...
+On Windows, OpenCL devices can be queried using `GPUCapsViewer`_, under the OpenCL tab. Note that the platform and device ID reported by this utility start from 0. So you should minus the reported IDs by 1.
```
Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: I think it's a typo and should be `start from 1`. At present it's possible to have `-1` id.

File: docs/GPU-Targets.rst [hunk 3]
```diff
@@ -69,75 +72,9 @@ We provide test R code below, but you can use the language of your choice with t
...
                        early_stopping_rounds = 10)
 
-Using a bad ``gpu_device_id`` is not critical, as it will fallback to:
-
--  ``gpu_device_id = 0`` if using ``gpu_platform_id = 0``
--  ``gpu_device_id = 1`` if using ``gpu_platform_id = 1``
-
-However, using a bad combination of ``gpu_platform_id`` and ``gpu_device_id`` will lead to a **crash** (you will lose your entire session content).
-Beware of it.
+Make sure you list the OpenCL devices in your system and set ``gpu_platform_id`` and ``gpu_device_id`` correctly. In the following examples, our system has 1 GPU platform (``gpu_platform_id = 0``) from AMD APP SDK. The first device ``gpu_device_id = 0`` is a GPU Device (AMD Oland), and the second device ``gpu_device_id = 1`` is the x86 CPU backend.
```
Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: "is a GPU Device (AMD Oland)" -> "is a GPU `d`evice (AMD Oland)"

File: docs/GPU-Targets.rst [hunk 4]
```diff
@@ -172,7 +109,9 @@ Example of GPU (``gpu_platform_id = 0``, ``gpu_device_id = 0``):
     [LightGBM] [Info] Trained a tree with leaves=7 and max_depth=5
     [2]:    test's rmse:0
 
-Example of CPU (``gpu_platform_id = 0``, ``gpu_device_id = 1``):
+Running on OpenCL CPU backend devices are in generally slow, and we observe crashes on some Windows and macOS systems. Make sure you check the ``Using GPU Device`` line in the log and it is not using a CPU. The above log shows that we are using ``Oland`` GPU From AMD and not CPU.
```
Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: "Oland GPU From AMD and not CPU." -> "Oland GPU `f`rom AMD and not CPU."

File: docs/GPU-Targets.rst [hunk 5]
```diff
@@ -207,90 +146,23 @@ Example of CPU (``gpu_platform_id = 0``, ``gpu_device_id = 1``):
...
+.. _ROCm: https://rocm.github.io/
+
 .. _our GitHub repo: https://github.com/Microsoft/LightGBM/releases/download/v2.0.12/AMD-APP-SDKInstaller-v3.0.130.136-GA-linux64.tar.bz2
 
 .. _NVIDIA CUDA Toolkit: https://developer.nvidia.com/cuda-downloads
 
 .. _clinfo: https://github.com/Oblomov/clinfo
 
 .. _GPUCapsViewer: http://www.ozone3d.net/gpu_caps_viewer/
+
```
Comment (StrikerRUS) [inline docs/GPU-Targets.rst]: Excess blank line.

**Comment by Laurae2** [comment]
@huanzhang12 I'll review after you went through @StrikerRUS review (in my case, mostly about the remaining typos).

Otherwise the content seems correct

**Comment by chivee** [comment]
ping @huanzhang12

**Comment by huanzhang12** [comment]
@Laurae2 @StrikerRUS @chivee Sorry for the delay! I have fixed the typos. Thank you @StrikerRUS  for checking the docs carefully and give good suggestions! @Laurae2 Let me know your comments on this, thanks!

**Comment by guolinke** [comment]
ping @Laurae2

**Pull request merged**